### PR TITLE
Bug 1881068: bump(k8s): Fix bug in reflector not recovering from 'Too large resource version'

### PR DESCRIPTION
**Pull request by gabemontero**
This is a v1.18 only k8s bug that is being pulled into 4.5.z

As the bug was already handled by the version of v1.19 pulled into 4.6/master, a cherry pick of a vendor dependency change from that release did not make sense.

/assign @adambkaplan

### CODE REVIEW

File: vendor/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 1]
```diff
@@ -873,6 +873,9 @@ const (
 	// FieldManagerConflict is used to report when another client claims to manage this field,
 	// It should only be returned for a request using server-side apply.
 	CauseTypeFieldManagerConflict CauseType = "FieldManagerConflict"
+	// CauseTypeResourceVersionTooLarge is used to report that the requested resource version
+	// is newer than the data observed by the API server, so the request cannot be served.
+	CauseTypeResourceVersionTooLarge CauseType = "ResourceVersionTooLarge"
```
Comment (gabemontero) [inline vendor/k8s.io/apimachinery/pkg/apis/meta/v1/types.go:878]: proof that we are pulling in the upstream k8s PR

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Comment by gabemontero** [comment]
previous image-eco failure looks unrelated to this change ... throttling related or some such

**Pull request merged**